### PR TITLE
feat: add swiss-map example

### PR DESCRIPTION
**Pull request by rfyiamcool**
### change

- update example in readme.
- add base.go to example directory.

### golang playground:

https://goplay.tools/snippet/CCn65XIj610

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,30 +7,73 @@ Uses [dolthub/swiss](https://github.com/dolthub/swiss) map implementation under
...
+		csmap.WithShardCount[string, int](32),
+
+		// if don't set custom hasher, use the built-in maphash.
+		csmap.WithCustomHasher[string, int](func(key string) uint64 {
+			hash := fnv.New64a()
+			hash.Write([]byte(key))
+			return hash.Sum64()
+		}),
+
+		// set the default capacity for each map. the default value is 0.
```
Comment (mhmtszr) [inline README.md:40]: Actually, this size is the total capacity of shard maps. We divide this value into shard counts to find the capacity for each map.

Comment (rfyiamcool) [reply README.md]: done, please cr again.

Comment (mhmtszr) [reply README.md]: @rfyiamcool I think it is still misleading what about 

// set the total capacity, every shard map has total capacity/shard count capacity. the default value is 0.

Comment (rfyiamcool) [reply README.md]: You are right, my comment was expressed incorrectly. 😁

**Comment by mhmtszr** [comment]
@rfyiamcool thanks for the contribution.

**Pull request merged**